lib/pbench/cli/agent/commands/triggers/register.py

```diff
@@ -0,0 +1,108 @@
+"""
+pbench-register-tool-trigger
+
+The sole purpose of this script is to register tool triggers for a given
+tool group of your choosing.  During the execution of a benchmark, the
+output of the benchmark is used to trigger the starting of tools and the
+stopping of tools.
+
+For a list of performance tools, look at the ${pbench_bin}/tool-scripts
+directory.
+
+"""
+import sys
+
+import click
+
+from pbench.agent.utils import setup_logging
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.triggers.base import TriggerCommand
+from pbench.cli.agent.options import common_options
+
+
+class TriggerRegister(TriggerCommand):
+    """Register tool trigger"""
+
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.logger = setup_logging()
+
+    def execute(self):
+        if self.verify_tool_group(self.context.group) != 0:
+            return 1
+
+        if ":" in self.context.start:
+            self.logger.error(
+                '%s: the start trigger cannot have a colon in it: "%s"',
+                self.name,
+                self.context.start,
+            )
+            return 1
+        if ":" in self.context.stop:
+            self.logger.error(
+                '%s: the stop trigger cannot have a colon in it: "%s"',
+                self.name,
+                self.context.stop,
+            )
+            return 1
+
+        # Remember this trigger
+        trigger = self.tool_group_dir / "__trigger__"
+        trigger.write_text("%s:%s\n" % (self.context.start, self.context.stop))
```

Comment:
Ah; should have looked at this first. So if a trigger file only has one line, the cat vs join in list-triggers doesn't matter. (Though it seems that joining lines may be a bit pointless in that case.)